src/ui/components/layouts/base_layout.rs

```diff
@@ -2,28 +2,49 @@ use crate::{
   contexts::theme_resource_context::ThemeResource,
   ui::components::{
     common::unpack::Unpack,
-    layouts::base_layout::{bottom_nav::BottomNav, top_nav::TopNav},
+    layouts::base_layout::{mobile_nav::MobileNav, side_nav::SideNav, top_nav::TopNav},
   },
 };
 use leptos::*;
 use leptos_meta::Html;
 use leptos_router::Outlet;
 
-mod bottom_nav;
+mod mobile_nav;
+mod side_nav;
 mod top_nav;
 
 #[component]
 pub fn BaseLayout() -> impl IntoView {
   let theme = expect_context::<ThemeResource>();
 
   view! {
-    <Transition>
-      <Unpack item=theme let:theme>
-        <Html attr:data-theme=theme/>
-        <TopNav/>
-        <Outlet/>
-        <BottomNav/>
-      </Unpack>
-    </Transition>
+    <div class="drawer h-full">
+      <input id="mobile-drawer" type="checkbox" class="drawer-toggle"/>
```

Comment:
The checkbox hack is how DaisyUI handles its drawers. Not very accessible sadly: I think the new popover API might be useful for making a drawer, but will have to try that out once it has wider browser support.